File: 100-Days-Series/Day 74/SearchInsertPosition.js
// Function to Search Insert Position
const getSearchInsertPosition=(nums,target)=>{
    let low=0;
    let high=nums.length-1;
    if(target>nums[target])
        return high+1;
    while(low<=high){
        let mid=Math.floor((low+high)/2);
        if(target==nums[mid])
            return mid;
        else if(target<nums[mid]){
            high=mid-1;
        }else{
            low=mid+1;
        }
    }
    return low;
}
// Let's Start Here
const nums=[1,2,5,6];
let target=4;
let pos=getSearchInsertPosition(nums,target);
console.log('Position is '+pos);

File: 100-Days-Series/Day 74/SearchInsertPosition.py

# Function to Search Insert Position
def get_search_insert_position(nums,target):
    low,high=0,len(nums)-1
    if target>nums[high]:
        return high+1
    while low<=high:
        mid=(low+high)//2
        if target==nums[mid]:
            return mid
        elif target<nums[mid]:
            high=mid-1
        else:
            low=mid+1
    return low

# Let's Start Here
nums=[1,3,5,6]
target=3
pos=get_search_insert_position(nums,target)
print('Position is ',pos)
